boards/native/Makefile.include

```diff
@@ -52,6 +52,9 @@ ifeq ($(OS_ARCH),x86_64)
 endif
 ifeq ($(OS),FreeBSD)
   ifeq ($(OS_ARCH),amd64)
+    ifeq ($(DISABLE_LIBSTDCPP),1)
+      LINKFLAGS += -nodefaultlibs -lgcc_s -lc -lc++ -lm -lpthread
+    endif
     LINKFLAGS += -m32 -DCOMPAT_32BIT -L/usr/lib32 -B/usr/lib32
   endif
```

Comment:
If this is only for `FreeBSD` then keep that ifdef as well.

```suggestion
ifeq (,$(filter libstdcpp,$(FEATURES_USED)))
   ifneq (,$(filter cpp,$(FEATURES_USED)))
      LINKFLAGS += -nodefaultlibs -lgcc_s -lc -lc++ -lm -lpthread
    endif
endif
```

Reply:
This would break on ATmega, as `-lpthread` is not provided there. This is just a very specific workaround for the very case a toolchain actually has a libstdc++ and that is used by default, but not usuable. The best fix for this would be the FreeBSD guys fixing their toolchain. In that case we could just drop this alltogether.

Normal platforms without libstdc++ (like AVR) do not try to link libstdc++ in, as they are explicitly build without support for libstdc++. For such platforms we don't need to mess with special linker flags and only need to pull in the `cxx_ctor_guards` module.

Reply:
Ok so ?:

```
ifneq (FreeBSD amd64, $(OS) $(OS_ARCH))
   ifneq (,$(filter cpp,$(FEATURES_USED)))
      # libstdc++ on FreeBSD is broken (does not work with -m32), disable it
      LINKFLAGS += -nodefaultlibs -lgcc_s -lc -lc++ -lm -lpthread
    endif
endif
```

Reply:
That would work (when the first `ifneq` is replaced by `ifeq`). But I would like there to be an override, in case a FreeBSD user has compiled her/his own toolchain with functioning multiarch support.

The `DISABLE_LIBSTDCPP` was there only so that a FreeBSD user can add `export DISABLE_LIBSTDCPP=0` to their `.profile` / `.bashrc`, in case they do want to use libstdc++ and have a working toolchain.

How about this instead:

```
ifeq (FreeBSD amd64, $(OS) $(OS_ARCH))
  # libstdc++ on FreeBSD is broken (does not work with -m32), disabling it unless user overwrites with `export DISABLE_LIBSTDCPP=0`
  DISABLE_LIBSTDCPP ?= 1
endif
ifeq (1,$(DISABLE_LIBSTDCPP))
  LINKFLAGS += -nodefaultlibs -lgcc_s -lc -lc++ -lm -lpthread
endif 
```

Reply:
Let's not forget that we are talking about a `native` specific file here, so breaking ATmega is not on the table ;-).

Reply:
Unless we move this to the global `Makefile.dep`. Unless I'm confused, this was part of the suggestion?

Reply:
I don't think so. Having `LINKFLAGS` set in `Makefile.dep` would be very untypical.

Reply:
Especially since toolchain-related stuff is very board/cpu dependent.

Reply:
OK. Sorry for the noise then!

Reply:
Done as @fjmolinas proposed (just with the FreeBSD/amd64 stuff kept in)